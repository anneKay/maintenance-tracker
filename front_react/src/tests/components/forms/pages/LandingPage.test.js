import React from 'react';
import { shallow } from 'enzyme';
import Landing from '../../../../components/pages/LandingPage';


describe(' Landing Page component rendering', () => {
  const wrapper = shallow(<Landing />);

  it('Should render properly', () => {
    expect(wrapper.length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });
});
