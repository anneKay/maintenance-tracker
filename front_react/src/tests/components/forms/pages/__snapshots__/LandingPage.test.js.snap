// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Landing Page component rendering Should render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LandingPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header
          className="header-primary"
          headerName=""
          id=""
          navOne="Contact"
          navThree="Signup"
          navTwo="Login"
          pathOne="/signup"
          pathThree="/signup"
          pathTwo="/login"
          pathname="/"
        />,
        <main>
          <section
            className="body-section"
          >
            <div
              className="container"
            >
              <h1>
                Maintenance Tracker
              </h1>
              <h3>
                Repair, Maintain and Monitor 
              </h3>
              <h3>
                your machineries with just one click
              </h3>
              <a
                href="../html/signup.html"
              >
                <button
                  className=""
                  name="button"
                  type="button"
                >
                   GET STARTED
                </button>
              </a>
            </div>
          </section>
        </main>,
        <Footer />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "header-primary",
          "headerName": "",
          "id": "",
          "navOne": "Contact",
          "navThree": "Signup",
          "navTwo": "Login",
          "pathOne": "/signup",
          "pathThree": "/signup",
          "pathTwo": "/login",
          "pathname": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <section
            className="body-section"
          >
            <div
              className="container"
            >
              <h1>
                Maintenance Tracker
              </h1>
              <h3>
                Repair, Maintain and Monitor 
              </h3>
              <h3>
                your machineries with just one click
              </h3>
              <a
                href="../html/signup.html"
              >
                <button
                  className=""
                  name="button"
                  type="button"
                >
                   GET STARTED
                </button>
              </a>
            </div>
          </section>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="container"
            >
              <h1>
                Maintenance Tracker
              </h1>
              <h3>
                Repair, Maintain and Monitor 
              </h3>
              <h3>
                your machineries with just one click
              </h3>
              <a
                href="../html/signup.html"
              >
                <button
                  className=""
                  name="button"
                  type="button"
                >
                   GET STARTED
                </button>
              </a>
            </div>,
            "className": "body-section",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1>
                  Maintenance Tracker
                </h1>,
                <h3>
                  Repair, Maintain and Monitor 
                </h3>,
                <h3>
                  your machineries with just one click
                </h3>,
                <a
                  href="../html/signup.html"
                >
                  <button
                    className=""
                    name="button"
                    type="button"
                  >
                     GET STARTED
                  </button>
                </a>,
              ],
              "className": "container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Maintenance Tracker",
                },
                "ref": null,
                "rendered": "Maintenance Tracker",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Repair, Maintain and Monitor ",
                },
                "ref": null,
                "rendered": "Repair, Maintain and Monitor ",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "your machineries with just one click",
                },
                "ref": null,
                "rendered": "your machineries with just one click",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <button
                    className=""
                    name="button"
                    type="button"
                  >
                     GET STARTED
                  </button>,
                  "href": "../html/signup.html",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " GET STARTED",
                    "className": "",
                    "name": "button",
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": " GET STARTED",
                  "type": "button",
                },
                "type": "a",
              },
            ],
            "type": "div",
          },
          "type": "section",
        },
        "type": "main",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header
            className="header-primary"
            headerName=""
            id=""
            navOne="Contact"
            navThree="Signup"
            navTwo="Login"
            pathOne="/signup"
            pathThree="/signup"
            pathTwo="/login"
            pathname="/"
          />,
          <main>
            <section
              className="body-section"
            >
              <div
                className="container"
              >
                <h1>
                  Maintenance Tracker
                </h1>
                <h3>
                  Repair, Maintain and Monitor 
                </h3>
                <h3>
                  your machineries with just one click
                </h3>
                <a
                  href="../html/signup.html"
                >
                  <button
                    className=""
                    name="button"
                    type="button"
                  >
                     GET STARTED
                  </button>
                </a>
              </div>
            </section>
          </main>,
          <Footer />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "header-primary",
            "headerName": "",
            "id": "",
            "navOne": "Contact",
            "navThree": "Signup",
            "navTwo": "Login",
            "pathOne": "/signup",
            "pathThree": "/signup",
            "pathTwo": "/login",
            "pathname": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <section
              className="body-section"
            >
              <div
                className="container"
              >
                <h1>
                  Maintenance Tracker
                </h1>
                <h3>
                  Repair, Maintain and Monitor 
                </h3>
                <h3>
                  your machineries with just one click
                </h3>
                <a
                  href="../html/signup.html"
                >
                  <button
                    className=""
                    name="button"
                    type="button"
                  >
                     GET STARTED
                  </button>
                </a>
              </div>
            </section>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="container"
              >
                <h1>
                  Maintenance Tracker
                </h1>
                <h3>
                  Repair, Maintain and Monitor 
                </h3>
                <h3>
                  your machineries with just one click
                </h3>
                <a
                  href="../html/signup.html"
                >
                  <button
                    className=""
                    name="button"
                    type="button"
                  >
                     GET STARTED
                  </button>
                </a>
              </div>,
              "className": "body-section",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1>
                    Maintenance Tracker
                  </h1>,
                  <h3>
                    Repair, Maintain and Monitor 
                  </h3>,
                  <h3>
                    your machineries with just one click
                  </h3>,
                  <a
                    href="../html/signup.html"
                  >
                    <button
                      className=""
                      name="button"
                      type="button"
                    >
                       GET STARTED
                    </button>
                  </a>,
                ],
                "className": "container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Maintenance Tracker",
                  },
                  "ref": null,
                  "rendered": "Maintenance Tracker",
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Repair, Maintain and Monitor ",
                  },
                  "ref": null,
                  "rendered": "Repair, Maintain and Monitor ",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "your machineries with just one click",
                  },
                  "ref": null,
                  "rendered": "your machineries with just one click",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <button
                      className=""
                      name="button"
                      type="button"
                    >
                       GET STARTED
                    </button>,
                    "href": "../html/signup.html",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": " GET STARTED",
                      "className": "",
                      "name": "button",
                      "type": "button",
                    },
                    "ref": null,
                    "rendered": " GET STARTED",
                    "type": "button",
                  },
                  "type": "a",
                },
              ],
              "type": "div",
            },
            "type": "section",
          },
          "type": "main",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
